In [1]:
import sys
import os

project_root = os.path.abspath('..')
if project_root not in sys.path:
    sys.path.append(project_root)

%load_ext autoreload
%autoreload 2


In [2]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

import warnings

from src.features import build_features

warnings.filterwarnings("ignore")
sns.set_style()
plt.rcParams["figure.figsize"] = (12, 6)

HOSPITAL_DEL_TORAX = 112103

## 1. Análisis de Procedimientos a Nivel País

En este análisis se quieren responder las siguiente preguntas:

1. ¿Cómo es el ranking de los hospitales para cada uno de los procedimientos realizads?
2. ¿Quién es el Hospital que realiza más de X procedimientos?
3. ¿El procedimiento X es realizado en todo el país? ¿En qué partes en específico se realiza?

Para responder esta pregunta se utilizará la base de datos de Egresos de Hospitales adscritos
a GRD entre 2019 y 2021.


In [3]:
vars_procedimientos = {f"PROCEDIMIENTO{i}": str for i in range(1, 31)}

In [4]:
cie_9 = pd.read_excel("../data/external/CIE-9.xlsx", dtype={"Código": str})
cie_10 = pd.read_excel("../data/external/CIE-10.xlsx")
esquema_hospitales = pd.read_excel(
    "../data/external/Esquema_Registro-2023.xls", sheet_name=4, header=7
).dropna(how="all", axis=1)

In [5]:
df_procesada = pd.read_csv("../data/processed/df_procesada.csv", sep=";", dtype=vars_procedimientos)

In [6]:
display(df_procesada.head(10))

COD_HOSPITAL  CIP_ENCRIPTADO    SEXO FECHA_NACIMIENTO  ETNIA   PROVINCIA  \
0        118100       1314867.0   MUJER       1988-06-19  OTRO   CONCEPCION   
1        118100       1354418.0   MUJER       1988-03-27  OTRO   CONCEPCION   
2        118100        239861.0   MUJER       2017-06-17  OTRO   CONCEPCION   
3        118100        330326.0  HOMBRE       2017-12-04  OTRO   CONCEPCION   
4        118100       1369293.0   MUJER       1995-12-06  OTRO   CONCEPCION   
5        118100        469385.0   MUJER       1942-06-29  OTRO   CONCEPCION   
6        118100       1146635.0   MUJER       1951-07-14  OTRO   CONCEPCION   
7        118100         93557.0  HOMBRE       1938-03-04  OTRO   CONCEPCION   
8        118100        674943.0   MUJER       1962-08-21  OTRO   CONCEPCION   
9        118100       1079955.0  HOMBRE       1944-12-07  OTRO   CONCEPCION   

                COMUNA NACIONALIDAD PREVISION SERVICIO_SALUD  ...  \
0  SAN PEDRO DE LA PAZ        CHILE  FONASA A     CONCEPCIÓN  ...   
1  SAN PEDRO DE LA PAZ        CHILE  FONASA D     CONCEPCIÓN  ...   
2           CONCEPCIÓN        CHILE  FONASA B     CONCEPCIÓN  ...   
3           CONCEPCIÓN        CHILE  FONASA A     CONCEPCIÓN  ...   
4              FLORIDA        CHILE  FONASA C     CONCEPCIÓN  ...   
5          CHIGUAYANTE        CHILE  FONASA B     CONCEPCIÓN  ...   
6           CONCEPCIÓN        CHILE  FONASA B     CONCEPCIÓN  ...   
7                 LOTA        CHILE  FONASA D     CONCEPCIÓN  ...   
8          SANTA JUANA        CHILE  FONASA B     CONCEPCIÓN  ...   
9           CONCEPCIÓN        CHILE  FONASA B     CONCEPCIÓN  ...   

  IR_29301_MORTALIDAD                         HOSPPROCEDENCIA ESTANCIA  \
0                 1.0                                     NaN      5.0   
1                 1.0                                     NaN      4.0   
2                 2.0                                     NaN      6.0   
3                 2.0                                     NaN      2.0   
4                 1.0                                     NaN      6.0   
5                 3.0                                     NaN     47.0   
6                 3.0                                     NaN     43.0   
7                 1.0                                     NaN     41.0   
8                 2.0  HOSPITAL CLORINDA AVELLO (SANTA JUANA)     33.0   
9                 3.0                                     NaN     31.0   

  ANIO_EGRESO MES_EGRESO   FECHA RANGO_ETARIO cod_comuna codregion  \
0      2019.0        3.0  2019-3     (30, 40]       8108         8   
1      2019.0        3.0  2019-3     (30, 40]       8108         8   
2      2019.0        3.0  2019-3      (0, 10]       8101         8   
3      2019.0        3.0  2019-3      (0, 10]       8101         8   
4      2019.0        3.0  2019-3     (20, 30]       8104         8   
5      2019.0        3.0  2019-3     (70, 80]       8103         8   
6      2019.0        3.0  2019-3     (60, 70]       8101         8   
7      2019.0        3.0  2019-3     (80, 90]       8106         8   
8      2019.0        3.0  2019-3     (50, 60]       8109         8   
9      2019.0        3.0  2019-3     (70, 80]       8101         8   

  Nombre Región  
0    del Biobío  
1    del Biobío  
2    del Biobío  
3    del Biobío  
4    del Biobío  
5    del Biobío  
6    del Biobío  
7    del Biobío  
8    del Biobío  
9    del Biobío  

[10 rows x 137 columns]

Los procedimientos en esta base de datos se encuentran en las columnas "PROCEDIMIENTOSX", donde X
va desde 1 hasta 30. Para analizar esta base de forma más expedita, se transformará de formato
ancho a formato largo, utilizando como identificador a las columnas ["COD_HOSPITAL", "ANIO_EGRESO",
"CIP_ENCRIPTADO", "DIAGNOSTICO1"]. Cabe destacar que la columna "CIP_ENCRIPTADO" hace referencia al
RUT del paciente.


In [7]:
procedimientos_formato_long = pd.melt(
    df_procesada,
    id_vars=[
        "COD_HOSPITAL",
        "ANIO_EGRESO",
        "TIPO_ACTIVIDAD",
        "CIP_ENCRIPTADO",
        "TIPOALTA",
        "DIAGNOSTICO1",
        "IR_29301_PESO",
        "IR_29301_SEVERIDAD",
    ],
    value_vars=vars_procedimientos.keys(),
    value_name="procedimiento",
)

In [8]:
var_categorica_a_contar = "procedimiento"
desglose_a_obtener = ["ANIO_EGRESO", "COD_HOSPITAL"]
grupo_ranking = ["ANIO_EGRESO", "procedimiento"]

conteo_y_ranking_procedimientos = (
    build_features.calcular_conteo_y_ranking_sobre_variable_categorica(
        procedimientos_formato_long,
        var_categorica_a_contar,
        desglose_a_obtener,
        grupo_ranking,
    )
)

In [9]:
display(conteo_y_ranking_procedimientos.head(20))

ANIO_EGRESO  COD_HOSPITAL procedimiento  conteo_procedimiento  \
180164       2021.0        113180         99.99                     4   
160709       2021.0        108100         99.99                     2   
148422       2021.0        103100         99.99                     1   
152362       2021.0        105100         99.99                     1   
166180       2021.0        110100         99.99                     1   
167426       2021.0        110120         99.99                     1   
179347       2021.0        113130         99.99                     1   
182873       2021.0        114101         99.99                     1   
187939       2021.0        115107         99.99                     1   
188727       2021.0        115110         99.99                     1   
197344       2021.0        118100         99.99                     1   
202517       2021.0        121109         99.99                     1   
206318       2021.0        122100         99.99                     1   
211715       2021.0        126100         99.99                     1   
213180       2021.0        129106         99.98                    47   
206567       2021.0        123100         99.98                    25   
162250       2021.0        109100         99.98                    16   
179984       2021.0        113180         99.98                    15   
165082       2021.0        110100         99.98                    14   
196030       2021.0        118100         99.98                    14   

        ranking_conteo_procedimiento  diferencias_procedimiento  \
180164                             1                          2   
160709                             2                          1   
148422                             3                          0   
152362                             4                          0   
166180                             5                          0   
167426                             6                          0   
179347                             7                          0   
182873                             8                          0   
187939                             9                          0   
188727                            10                          0   
197344                            11                          0   
202517                            12                          0   
206318                            13                          0   
211715                            14                        -46   
213180                             1                         22   
206567                             2                          9   
162250                             3                          1   
179984                             4                          1   
165082                             5                          0   
196030                             6                          0   

        totales_procedimiento  proporcion_procedimiento  
180164                     18                  0.222222  
160709                     18                  0.111111  
148422                     18                  0.055556  
152362                     18                  0.055556  
166180                     18                  0.055556  
167426                     18                  0.055556  
179347                     18                  0.055556  
182873                     18                  0.055556  
187939                     18                  0.055556  
188727                     18                  0.055556  
197344                     18                  0.055556  
202517                     18                  0.055556  
206318                     18                  0.055556  
211715                     18                  0.055556  
213180                    235                  0.200000  
206567                    235                  0.106383  
162250                    235                  0.068085  
179984                

El resultado muestra el ranking para todos los procedimientos asignados en el país en cada uno de
los años analizados (2019 a 2021).

A modo de ejemplo, se ilustra el ranking en el año 2021 para el procedimiento 99.99. En este caso,
se observa que los hospitales con código 108100 y 113180 están en los primeros lugares, realizando
2 procedimientos cada uno.

Ahora, si se filtra por el hospital del tórax, y sólo los procedimientos donde el hospital sea
ranking 1, entonces se obtendrán procedimientos altamente relevantes para el país.


In [10]:
conteo_y_ranking_procedimientos = conteo_y_ranking_procedimientos.merge(
    cie_9, how="left", left_on="procedimiento", right_on="Código"
).merge(
    esquema_hospitales, how="left", left_on="COD_HOSPITAL", right_on="Código nuevo Establecimiento"
)


In [11]:
display(conteo_y_ranking_procedimientos)

ANIO_EGRESO  COD_HOSPITAL procedimiento  conteo_procedimiento  \
0            2021.0        113180         99.99                     4   
1            2021.0        108100         99.99                     2   
2            2021.0        103100         99.99                     1   
3            2021.0        105100         99.99                     1   
4            2021.0        110100         99.99                     1   
...             ...           ...           ...                   ...   
214719       2019.0        105101         00.01                    65   
214720       2019.0        106100         00.01                    13   
214721       2019.0        113100         00.01                     1   
214722       2019.0        120101         00.01                     1   
214723       2019.0        114105             0                     2   

        ranking_conteo_procedimiento  diferencias_procedimiento  \
0                                  1                          2   
1                                  2                          1   
2                                  3                          0   
3                                  4                          0   
4                                  5                          0   
...                              ...                        ...   
214719                             1                         52   
214720                             2                         12   
214721                             3                          0   
214722                             4                         -1   
214723                             1                       <NA>   

        totales_procedimiento  proporcion_procedimiento Código  \
0                          18                  0.222222  99.99   
1                          18                  0.111111  99.99   
2                          18                  0.055556  99.99   
3                          18                  0.055556  99.99   
4                          18                  0.055556  99.99   
...                       ...                       ...    ...   
214719                     80                  0.812500  00.01   
214720                     80                  0.162500  00.01   
214721                     80                  0.012500  00.01   
214722                     80                  0.012500  00.01   
214723                      2                  1.000000    NaN   

                                         Descripción  ...  \
0                      PROCEDIMIENTO MISCELANEO.OTRO  ...   
1                      PROCEDIMIENTO MISCELANEO.OTRO  ...   
2                      PROCEDIMIENTO MISCELANEO.OTRO  ...   
3                      PROCEDIMIENTO MISCELANEO.OTRO  ...   
4                      PROCEDIMIENTO MISCELANEO.OTRO  ...   
...                                              ...  ...   
214719  ULTRASONIDO TERAPEUTICO.VASO CABEZA Y CUELLO  ...   
214720  ULTRASONIDO TERAPEUTICO.VASO CABEZA Y CUELLO  ...   
214721  ULTRASONIDO TERAPEUTICO.VASO CABEZA Y CUELLO  ...   
214722  ULTRASONIDO TERAPEUTICO.VASO CABEZA Y CUELLO  ...   
214723                                           NaN  ...   

                                                 Capítulo  \
0       16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
1       16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
2       16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
3       16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
4       16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
...                                                   ...   
214719  00. (00 00) PROCEDIMIENTOS E INTERVENCIONES NO...   
214720  00. (00 00) PROCEDIMIENTOS E INTERVENCIONES NO...   
214721  00. (00 00) PROCEDIMIENTOS E INTERVENCIONES NO...   
214722  00. (00 00) PROCEDIMIENTOS E INTERVENCIONES NO...   
214723                                                NaN   

       Código nuevo Establecimiento  \
0                           

Ahora se puede observar el nombre del procedimiento analizado y el nombre del hospital visto.


In [12]:
conteo_procedimientos_torax = conteo_y_ranking_procedimientos.query(
    "COD_HOSPITAL == @HOSPITAL_DEL_TORAX"
).sort_values(["ANIO_EGRESO", "conteo_procedimiento"], ascending=False)

mejores_procedimientos_torax = conteo_procedimientos_torax.query(
    "ranking_conteo_procedimiento == 1"
)

display(mejores_procedimientos_torax)

ANIO_EGRESO  COD_HOSPITAL procedimiento  conteo_procedimiento  \
54292        2021.0        112103         39.61                   626   
54248        2021.0        112103         39.64                   584   
52930        2021.0        112103         42.23                   391   
57290        2021.0        112103         37.22                   280   
13241        2021.0        112103         89.46                   278   
...             ...           ...           ...                   ...   
200772       2019.0        112103         34.89                     2   
201686       2019.0        112103         33.49                     2   
210994       2019.0        112103         07.81                     2   
198511       2019.0        112103         38.46                     1   
198560       2019.0        112103         38.42                     1   

        ranking_conteo_procedimiento  diferencias_procedimiento  \
54292                              1                         68   
54248                              1                        246   
52930                              1                        342   
57290                              1                         22   
13241                              1                         82   
...                              ...                        ...   
200772                             1                          1   
201686                             1                          1   
210994                             1                          1   
198511                             1                          0   
198560                             1                          0   

        totales_procedimiento  proporcion_procedimiento Código  \
54292                    3184                  0.196608  39.61   
54248                    2020                  0.289109  39.64   
52930                     607                  0.644152  42.23   
57290                     801                  0.349563  37.22   
13241                    1355                  0.205166  89.46   
...                       ...                       ...    ...   
200772                      6                  0.333333  34.89   
201686                     10                  0.200000  33.49   
210994                     10                  0.200000  07.81   
198511                      3                  0.333333  38.46   
198560                      5                  0.200000  38.42   

                                              Descripción  ...  \
54292   CIRCULACION EXTRACORPOREA PARA CIRUGIA CORAZON...  ...   
54248                  MARCAPASOS CORAZON INTRAOPERATORIO  ...   
52930                                  ESOFAGOSCOPIA OTRA  ...   
57290                  CATETERISMO CORAZON LADO IZQUIERDO  ...   
13241   COMPROBACION FORMA ONDA ARTEFACTO MARCAPASOS C...  ...   
...                                                   ...  ...   
200772                           OPERACION DIAFRAGMA.OTRA  ...   
201686        OPERACION PULMON REPARADORA Y PLASTICA OTRA  ...   
210994                         ESCISION TIMO PARCIAL OTRA  ...   
198511          RESECCION ARTERIA ABDOMEN CON SUSTITUCION  ...   
198560  RESECCION VASO CABEZA Y CUELLO CON SUSTITUCION...  ...   

                                                 Capítulo  \
54292   07. (35 39) OPERACIONES SOBRE EL APARATO CARDI...   
54248   07. (35 39) OPERACIONES SOBRE EL APARATO CARDI...   
52930   09. (42 54) OPERACIONES SOBRE EL APARATO DIGES...   
57290   07. (35 39) OPERACIONES SOBRE EL APARATO CARDI...   
13241   16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
...                                                   ...   
200772  06. (30 34) OPERACIONES SOBRE EL APARATO RESPI...   
201686  06. (30 34) OPERACIONES SOBRE EL APARATO RESPI...   
210994  02. (06 07) OPERACIONES SOBRE EL SISTEMA ENDOC...   
198511  07. (35 39) OPERACIONES SOBRE EL APARATO CARDI...   
198560  07. (35 39) OPERACIONES SOBRE EL APARATO CARDI...   

       

Los resultados muestran que el tórax en el año 2021 fue altamente relevante en los procedimientos
39.61, 39.64, 42.23, entre otros. Además, se puede observar la diferencia que tiene el Tórax
con los siguientes establecimientos de salud y el total de procedimientos realizados en el país.


Ahora, se quiere saber la cantidad de procedimientos, pero sabiendo si fue realizado de forma
ambulatoria, hospitalizada, cirugia mayor ambulatoria, etc...


In [13]:
var_categorica_a_contar = "procedimiento"
desglose_a_obtener = ["ANIO_EGRESO", "COD_HOSPITAL", "TIPO_ACTIVIDAD"]
grupo_ranking = ["ANIO_EGRESO", "TIPO_ACTIVIDAD", "procedimiento"]

conteo_y_ranking_procedimientos_por_tipoalta = (
    build_features.calcular_conteo_y_ranking_sobre_variable_categorica(
        procedimientos_formato_long,
        var_categorica_a_contar,
        desglose_a_obtener,
        grupo_ranking,
    )
)

In [14]:
conteo_y_ranking_procedimientos_por_tipoalta = conteo_y_ranking_procedimientos_por_tipoalta.merge(
    cie_9, how="left", left_on="procedimiento", right_on="Código"
).merge(
    esquema_hospitales, how="left", left_on="COD_HOSPITAL", right_on="Código nuevo Establecimiento"
)


In [15]:
with pd.ExcelWriter("../data/interim/procedimientos_grd.xlsx") as file:
    conteo_procedimientos_torax.to_excel(file, sheet_name="procedimientos_torax", index=False)
    conteo_y_ranking_procedimientos.to_excel(file, sheet_name="resumen_nacional", index=False)
    conteo_y_ranking_procedimientos_por_tipoalta.to_excel(
        file, sheet_name="resumen_nacional_tipoalta", index=False
    )

## 2. Obtención de perfiles de pacientes por procedimientos

En este apartado se quieren responder las siguientes preguntas:

1. ¿Cuál es la **mortalidad** de cada uno de los procedimientos?
2. ¿Cuál es el **peso GRD** promedio de cada uno de los procedimientos?
3. ¿Cuál es la **severidad** de cada uno de los procedimientos?
4. Dentro de los pacientes muertos, **¿Cuál es la severidad que tienen?**

Ahora, la base de GRD tiene hasta 30 procedimientos, por lo que en un mismo egreso se pueden
repetir procedimientos. Para reducir la complejidad del análisis, se utilizará solamente el primer
procedimiento para agrupar por severidad y mortalidad. Esto, ya que 1 procedimiento corresponderá
a 1 egreso.

Por lo tanto, la base de datos se debe mantener como está, pero solamente seleccionando las variables
recién mencionadas


In [16]:
display(procedimientos_formato_long)

COD_HOSPITAL  ANIO_EGRESO   TIPO_ACTIVIDAD  CIP_ENCRIPTADO  \
0               118100       2019.0  HOSPITALIZACIÓN       1314867.0   
1               118100       2019.0  HOSPITALIZACIÓN       1354418.0   
2               118100       2019.0  HOSPITALIZACIÓN        239861.0   
3               118100       2019.0  HOSPITALIZACIÓN        330326.0   
4               118100       2019.0  HOSPITALIZACIÓN       1369293.0   
...                ...          ...              ...             ...   
82471585        105100       2021.0  HOSPITALIZACIÓN      68882797.0   
82471586        120101       2021.0  HOSPITALIZACIÓN      97789022.0   
82471587        114101       2021.0  HOSPITALIZACIÓN      71164179.0   
82471588        107100       2021.0  HOSPITALIZACIÓN      78884679.0   
82471589        102100       2021.0  HOSPITALIZACIÓN      71716269.0   

                 TIPOALTA DIAGNOSTICO1  IR_29301_PESO  IR_29301_SEVERIDAD  \
0               DOMICILIO        O82.0         0.5744                 2.0   
1               DOMICILIO        O26.9         0.2951                 1.0   
2               DOMICILIO        K92.0         0.6736                 2.0   
3               DOMICILIO        A08.3         0.5475                 2.0   
4               DOMICILIO        O60.0         0.3107                 1.0   
...                   ...          ...            ...                 ...   
82471585        DOMICILIO        N39.0         1.6547                 3.0   
82471586        DOMICILIO        K40.9         0.8122                 1.0   
82471587        DOMICILIO        Z51.8         1.0867                 2.0   
82471588        DOMICILIO        K80.0         1.1472                 2.0   
82471589  ALTA VOLUNTARIA          N10         0.5701                 2.0   

                 variable procedimiento  
0          PROCEDIMIENTO1          74.1  
1          PROCEDIMIENTO1         88.78  
2          PROCEDIMIENTO1         87.03  
3          PROCEDIMIENTO1         90.92  
4          PROCEDIMIENTO1         75.34  
...                   ...           ...  
82471585  PROCEDIMIENTO30           NaN  
82471586  PROCEDIMIENTO30           NaN  
82471587  PROCEDIMIENTO30           NaN  
82471588  PROCEDIMIENTO30           NaN  
82471589  PROCEDIMIENTO30           NaN  

[82471590 rows x 10 columns]

Ahora que se tiene la base de datos con menos variables, se quiere responder a la pregunta:

¿Cuál es la mortalidad de cada uno de los procedimientos?


In [17]:
desglose_procedimientos = procedimientos_formato_long.groupby(
    [
        "COD_HOSPITAL",
        "ANIO_EGRESO",
        "TIPO_ACTIVIDAD",
        "TIPOALTA",
        "procedimiento",
        "IR_29301_SEVERIDAD",
    ]
).agg(cantidad_procedimientos=("procedimiento", "count"), peso_grd_medio=("IR_29301_PESO", "mean"))

display(desglose_procedimientos.head(10))

desglose_procedimientos = desglose_procedimientos.reset_index().sort_values(
    ["COD_HOSPITAL", "ANIO_EGRESO", "cantidad_procedimientos"]
)

cantidad_procedimientos  \
COD_HOSPITAL ANIO_EGRESO TIPO_ACTIVIDAD                  TIPOALTA  procedimiento IR_29301_SEVERIDAD                            
101100       2019.0      CIRUGÍA MAYOR AMBULATORIA (CMA) DOMICILIO 03.92         0.0                                      44   
                                                                   04.07         0.0                                       1   
                                                                   04.11         0.0                                       1   
                                                                   04.43         0.0                                      45   
                                                                   04.81         0.0                                      15   
                                                                   06.31         0.0                                       1   
                                                                   06.7          0.0                                       1   
                                                                   08.01         0.0                                       1   
                                                                   08.20         0.0                                       6   
                                                                   08.22         0.0                                       1   

                                                                                                     peso_grd_medio  
COD_HOSPITAL ANIO_EGRESO TIPO_ACTIVIDAD                  TIPOALTA  procedimiento IR_29301_SEVERIDAD                  
101100       2019.0      CIRUGÍA MAYOR AMBULATORIA (CMA) DOMICILIO 03.92         0.0                       0.148200  
                                                                   04.07         0.0                       1.216700  
                                                                   04.11         0.0                       0.297300  
                                                                   04.43         0.0                       0.304500  
                                                                   04.81         0.0                       0.147913  
                                                                   06.31         0.0                       0.987000  
                                                                   06.7          0.0                       0.552400  
                                                                   08.01         0.0                       0.188500  
                                                                   08.20         0.0                       0.329967  
                                                                   08.22         0.0                       0.273700

Los resultados anteriores indican lo siguiente:

El hospital 101100 en el año 2019 tuvo 44 procedimientos 1 03.92 que tuvieron una severidad de 0.
Estos procedimientos fueron realizados en una cirugía mayor ambulatorio (CMA), y la alta de
sus pacientes fueron en DOMICILIO. Dentro de los 44 egresos, estos tuvieron una complejidad
promedio de 0.148200.

Ahora, para obtener la mortalidad de cada procedimiento es necesario consultar los pacientes
que tienen un TIPOALTA igual a FALLECIDO.


In [18]:
display(desglose_procedimientos.query("TIPOALTA == 'FALLECIDO' and procedimiento == '00.66'"))

COD_HOSPITAL  ANIO_EGRESO   TIPO_ACTIVIDAD   TIPOALTA procedimiento  \
3496          101100       2019.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
3497          101100       2019.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
8207          101100       2020.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
12628         101100       2021.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
12629         101100       2021.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
...              ...          ...              ...        ...           ...   
784235        126100       2020.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
788411        126100       2021.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
797134        129100       2019.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
799585        129100       2020.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
816426        133150       2020.0  HOSPITALIZACIÓN  FALLECIDO         00.66   

        IR_29301_SEVERIDAD  cantidad_procedimientos  peso_grd_medio  
3496                   2.0                        2        1.818300  
3497                   3.0                        3        2.484000  
8207                   3.0                        3        3.048567  
12628                  1.0                        2        1.718400  
12629                  3.0                        7        2.725957  
...                    ...                      ...             ...  
784235                 3.0                        4        3.741600  
788411                 3.0                        5        4.995440  
797134                 3.0                        1        2.484000  
799585                 3.0                        2        2.112300  
816426                 3.0                        2        1.598750  

[197 rows x 8 columns]

Los resultados anteriores indican que en el Hospital 101100 en el año 2021, 7 de los procedimientos
00.66 tuvieron fallecimientos en una hospitalización. Estos procedimientos tenían una severidad de 3,
con un peso GRD promedio de 2.7260.

A modo de ejemplo, se realizará la consulta en la base de datos y se mostrarán los pacientes
que cumplen los criterios.


In [19]:
consulta_inicial = df_procesada.query(
    "ANIO_EGRESO == 2021.0 and COD_HOSPITAL == 101100 and "
    "IR_29301_SEVERIDAD == 3.0 and TIPOALTA == 'FALLECIDO'"
)

filas_con_procedimiento = (
    consulta_inicial[vars_procedimientos.keys()]
    .apply(lambda x: x.str.contains("00.66"))
    .sum(axis=1)
    .astype(bool)
)

display(consulta_inicial[filas_con_procedimiento])

COD_HOSPITAL  CIP_ENCRIPTADO    SEXO FECHA_NACIMIENTO    ETNIA  \
2101790        101100      78353743.0  HOMBRE       1949-06-20  NINGUNO   
2495416        101100      70827726.0   MUJER       1946-04-20  NINGUNO   
2513407        101100      70124532.0   MUJER       1954-09-08  NINGUNO   
2536108        101100      68419768.0  HOMBRE       1957-01-26  NINGUNO   
2592167        101100      70259268.0   MUJER       1945-10-18  NINGUNO   
2675921        101100      71418919.0   MUJER       1970-02-17  NINGUNO   
2708809        101100      68127172.0  HOMBRE       1939-08-01  NINGUNO   

        PROVINCIA COMUNA NACIONALIDAD PREVISION SERVICIO_SALUD  ...  \
2101790     ARICA  ARICA        CHILE  FONASA B          ARICA  ...   
2495416     ARICA  ARICA        CHILE  FONASA C          ARICA  ...   
2513407     ARICA  ARICA        CHILE  FONASA A          ARICA  ...   
2536108     ARICA  ARICA        CHILE  FONASA B          ARICA  ...   
2592167     ARICA  ARICA        CHILE  FONASA B          ARICA  ...   
2675921     ARICA  ARICA        CHILE  FONASA B          ARICA  ...   
2708809     ARICA  ARICA        CHILE  FONASA C          ARICA  ...   

        IR_29301_MORTALIDAD HOSPPROCEDENCIA ESTANCIA ANIO_EGRESO MES_EGRESO  \
2101790                 3.0             NaN      2.0      2021.0        4.0   
2495416                 3.0             NaN     12.0      2021.0       10.0   
2513407                 3.0             NaN      1.0      2021.0        2.0   
2536108                 3.0             NaN      5.0      2021.0        6.0   
2592167                 3.0             NaN      2.0      2021.0       11.0   
2675921                 3.0             NaN     10.0      2021.0       12.0   
2708809                 3.0             NaN      3.0      2021.0        4.0   

           FECHA RANGO_ETARIO cod_comuna codregion          Nombre Región  
2101790   2021-4     (70, 80]      15101        15  de Arica y Parinacota  
2495416  2021-10     (70, 80]      15101        15  de Arica y Parinacota  
2513407   2021-2     (60, 70]      15101        15  de Arica y Parinacota  
2536108   2021-6     (60, 70]      15101        15  de Arica y Parinacota  
2592167  2021-11     (70, 80]      15101        15  de Arica y Parinacota  
2675921  2021-12     (50, 60]      15101        15  de Arica y Parinacota  
2708809   2021-4     (80, 90]      15101        15  de Arica y Parinacota  

[7 rows x 137 columns]

Sin embargo, es necesario saber cuánto es el porcentaje de muertos del total.


In [20]:
total_proceds = desglose_procedimientos.groupby(["COD_HOSPITAL", "ANIO_EGRESO", "procedimiento"])[
    "cantidad_procedimientos"
].transform("sum")

desglose_procedimientos["total_de_procedimientos"] = total_proceds

desglose_procedimientos["ratio_de_procedimientos"] = round(
    desglose_procedimientos["cantidad_procedimientos"]
    / desglose_procedimientos["total_de_procedimientos"],
    2,
)

In [21]:
desglose_procedimientos.query("COD_HOSPITAL == 101100 and ANIO_EGRESO == 2019.0 "
                              "and procedimiento == '00.66'")

COD_HOSPITAL  ANIO_EGRESO   TIPO_ACTIVIDAD   TIPOALTA procedimiento  \
1234        101100       2019.0  HOSPITALIZACIÓN  DOMICILIO         00.66   
3496        101100       2019.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
3497        101100       2019.0  HOSPITALIZACIÓN  FALLECIDO         00.66   
1233        101100       2019.0  HOSPITALIZACIÓN  DOMICILIO         00.66   
1232        101100       2019.0  HOSPITALIZACIÓN  DOMICILIO         00.66   

      IR_29301_SEVERIDAD  cantidad_procedimientos  peso_grd_medio  \
1234                 3.0                        2          2.4840   
3496                 2.0                        2          1.8183   
3497                 3.0                        3          2.4840   
1233                 2.0                       12          1.8183   
1232                 1.0                       15          1.7184   

      total_de_procedimientos  ratio_de_procedimientos  
1234                       34                     0.06  
3496                       34                     0.06  
3497                       34                     0.09  
1233                       34                     0.35  
1232                       34                     0.44

Los resultados anteriores se pueden interpretar de la siguiente manera:

- El hospital 101100, en el año 2019 tuvo 5 procedimientos en donde
  la persona falleció realizandose un procedimiento 00.66. Estos egresos tenían una severidad 2
  y 3, y corresponden al 15% de todos los procedimientos realizados.


In [22]:
desglose_procedimientos = desglose_procedimientos.merge(
    cie_9, how="left", left_on="procedimiento", right_on="Código"
).merge(
    esquema_hospitales,
    how="left",
    left_on="COD_HOSPITAL",
    right_on="Código nuevo Establecimiento",
)

In [23]:
desglose_procedimientos.to_excel(
    "../data/interim/mortalidad_y_severidad_procedimientos.xlsx",
    index=False,
)


En el Hospital del Tórax existe un interés por saber cuál es la condición de ingreso de los
pacientes que se hicieron una angioplastia el 2019 y que fallecieron dentro de una hospitalizacion
diurna.

In [24]:
casos_angio = df_procesada.query(
    "ANIO_EGRESO == 2019.0 and COD_HOSPITAL == @HOSPITAL_DEL_TORAX and TIPOALTA == 'FALLECIDO' "
    "and TIPO_ACTIVIDAD == 'HOSPITALIZACIÓN DIURNA'"
)

filas_con_procedimiento = (
    casos_angio[vars_procedimientos.keys()]
    .apply(lambda x: x.str.contains("00.66"))
    .sum(axis=1)
    .astype(bool)
)

muertos_torax_angio = casos_angio[filas_con_procedimiento]

In [25]:
display(muertos_torax_angio[
    [
        "SEXO",
        "NACIONALIDAD",
        "PREVISION",
        "TIPO_PROCEDENCIA",
        "TIPO_INGRESO",
        "IR_29301_PESO",
        "IR_29301_SEVERIDAD",
        "HOSPPROCEDENCIA",
        "Nombre Región",
        "COMUNA",
        "RANGO_ETARIO"
    ]
])

SEXO NACIONALIDAD        PREVISION  \
401706   MUJER        CHILE         FONASA D   
464271   MUJER        CHILE         FONASA A   
480795  HOMBRE        CHILE         FONASA A   
499763   MUJER        CHILE         FONASA B   
499812  HOMBRE        CHILE         FONASA D   
512641  HOMBRE        CHILE         FONASA B   
513160  HOMBRE    ARGENTINA  NO IDENTIFICADA   
551734  HOMBRE        CHILE         FONASA A   
649982   MUJER        CHILE         FONASA A   
789463   MUJER        CHILE         FONASA B   
961339  HOMBRE        CHILE           ISAPRE   
977726  HOMBRE        CHILE         FONASA D   

                                         TIPO_PROCEDENCIA TIPO_INGRESO  \
401706  CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...   PROGRAMADA   
464271                      APS URGENCIA (SAPU, SUR, SUC)     URGENCIA   
480795                      APS URGENCIA (SAPU, SUR, SUC)     URGENCIA   
499763                      APS URGENCIA (SAPU, SUR, SUC)     URGENCIA   
499812  CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...     URGENCIA   
512641                         OTROS HOSPITALES DE LA RED     URGENCIA   
513160                         OTROS HOSPITALES DE LA RED     URGENCIA   
551734                         OTROS HOSPITALES DE LA RED     URGENCIA   
649982                      APS URGENCIA (SAPU, SUR, SUC)     URGENCIA   
789463                         OTROS HOSPITALES DE LA RED     URGENCIA   
961339                      APS URGENCIA (SAPU, SUR, SUC)     URGENCIA   
977726                      APS URGENCIA (SAPU, SUR, SUC)     URGENCIA   

        IR_29301_PESO  IR_29301_SEVERIDAD  \
401706         1.7114                 0.0   
464271         1.7114                 0.0   
480795         1.7114                 0.0   
499763         1.7114                 0.0   
499812         1.7114                 0.0   
512641         1.7114                 0.0   
513160         1.7114                 0.0   
551734         1.7114                 0.0   
649982         1.7114                 0.0   
789463         1.7114                 0.0   
961339         1.7114                 0.0   
977726         1.7114                 0.0   

                                         HOSPPROCEDENCIA  \
401706                                               NaN   
464271                                               NaN   
480795                                               NaN   
499763                                               NaN   
499812                                               NaN   
512641  HOSPITAL DR. LUIS TISNÉ B. (SANTIAGO, PEÑALOLÉN)   
513160     HOSPITAL DEL SALVADOR (SANTIAGO, PROVIDENCIA)   
551734     HOSPITAL DEL SALVADOR (SANTIAGO, PROVIDENCIA)   
649982                                               NaN   
789463     HOSPITAL DEL SALVADOR (SANTIAGO, PROVIDENCIA)   
961339                                               NaN   
977726                                               NaN   

                    Nombre Región        COMUNA RANGO_ETARIO  
401706  Metropolitana de Santiago         ÑUÑOA     (50, 60]  
464271  Metropolitana de Santiago     LA GRANJA     (50, 60]  
480795  Metropolitana de Santiago         ÑUÑOA     (50, 60]  
499763  Metropolitana de Santiago         ÑUÑOA     (60, 70]  
499812  Metropolitana de Santiago         ÑUÑOA     (70, 80]  
512641  Metropolitana de Santiago     PEÑALOLÉN     (70, 80]  
513160  Metropolitana de Santiago   PROVIDENCIA     (70, 80]  
551734  Metropolitana de Santiago   LA CISTERNA     (60, 70]  
649982              de Valparaíso  VIÑA DEL MAR     (80, 90]  
789463  Metropolitana de Santiago   PROVIDENCIA     (70, 80]  
961339  Metropolitana de Santiago         ÑUÑOA     (70, 80]  
977726  Metropolitana de Santiago    LA FLORIDA     (40, 50]

## 3. Análisis de Procedimientos 3

En este apartado se quieren identificar los rankings de procedimientos de alta relevancia/complejidad
(PROCEDIMIENTO1 hasta PROCEDIMIENTO3).

In [26]:
procedimientos_3_long = pd.melt(
    df_procesada,
    id_vars=[
        "COD_HOSPITAL",
        "ANIO_EGRESO",
        "TIPO_ACTIVIDAD",
        "CIP_ENCRIPTADO",
        "TIPOALTA",
        "DIAGNOSTICO1",
        "IR_29301_PESO",
        "IR_29301_SEVERIDAD",
    ],
    value_vars=["PROCEDIMIENTO1", "PROCEDIMIENTO2", "PROCEDIMIENTO3"],
    value_name="procedimiento",
)

In [27]:
var_categorica_a_contar = "procedimiento"
desglose_a_obtener = ["ANIO_EGRESO", "COD_HOSPITAL"]
grupo_ranking = ["ANIO_EGRESO", "procedimiento"]

conteo_y_ranking_procedimientos_3 = (
    build_features.calcular_conteo_y_ranking_sobre_variable_categorica(
        procedimientos_3_long,
        var_categorica_a_contar,
        desglose_a_obtener,
        grupo_ranking,
    )
)

In [28]:
conteo_y_ranking_procedimientos_3 = conteo_y_ranking_procedimientos_3.merge(
    cie_9, how="left", left_on="procedimiento", right_on="Código"
).merge(
    esquema_hospitales, how="left", left_on="COD_HOSPITAL", right_on="Código nuevo Establecimiento"
)

In [29]:
conteo_y_ranking_procedimientos_3.to_excel("../data/interim/procedimientos_3_grd.xlsx", index=False)

## 4. Obtención de canastas de procedimientos por diagnóstico

En este análisis se quieren responder las siguientes preguntas:

- ¿Cuántos procedimientos se le deben realizar a una persona que tenga X diagnóstico 
(Ej: Cáncer de Pulmón)?
- ¿Cuál es el procedimiento más realizado dentro de un diagnóstico?
- ¿En qué diagnósticos es más utilizado X procedimiento (Ej: Angioplastia)?

Esta pregunta se quiere responder únicamente para el Hospital del Tórax. Además, para tener un
análisis promedio se utilizaran los años 2019, 2020 y 2021 en conjunto para responder las preguntas
anteriores. Además, se utilizarán todos los tipo de actividad realizada 
(HOSPITALIZACIÓN, HOSPITALIZACIÓN DIURNA, CMA, etc).

In [8]:
procedimientos_torax_long = procedimientos_formato_long.query("COD_HOSPITAL == @HOSPITAL_DEL_TORAX")

In [9]:
DIAGNOSTICOS_MAS_RELEVANTES = [
    "C33",
    "C34.0",
    "C34.1",
    "C34.2",
    "C34.3",
    "C38.1",
    "C38.4",
    "C45.0",
    "C78.0",
    "C78.2",
    "D14.3",
    "D38.1",
    "E84.8",
    "I05.1",
    "I08.0",
    "I08.1",
    "I34.0",
    "I35.0",
    "I35.1",
    "I35.2",
    "I42.0",
    "I45.6",
    "I47.2",
    "I49.5",
    "I71.0",
    "I71.2",
    "J39.8",
    "J47",
    "J67.9",
    "J84.1",
    "J84.8",
    "J84.9",
    "J86.0",
    "J86.9",
    "J90",
    "J93.1",
    "J95.5",
    "J98.0",
    "M34.8",
    "Q21.1",
    "Q23.1",
    "Q67.6",
    "T82.0",
    "T82.1",
    "Z45.0",
]

In [13]:
display(procedimientos_torax_long)

COD_HOSPITAL  ANIO_EGRESO          TIPO_ACTIVIDAD  CIP_ENCRIPTADO  \
3946            112103       2019.0         HOSPITALIZACIÓN       1137216.0   
9385            112103       2019.0         HOSPITALIZACIÓN       1084360.0   
9442            112103       2019.0         HOSPITALIZACIÓN        822388.0   
9443            112103       2019.0  HOSPITALIZACIÓN DIURNA        317615.0   
9539            112103       2019.0         HOSPITALIZACIÓN       1397457.0   
...                ...          ...                     ...             ...   
82470182        112103       2021.0         HOSPITALIZACIÓN      75968176.0   
82470668        112103       2021.0         HOSPITALIZACIÓN      81034371.0   
82471078        112103       2021.0         HOSPITALIZACIÓN      67821076.0   
82471166        112103       2021.0         HOSPITALIZACIÓN      76198942.0   
82471581        112103       2021.0         HOSPITALIZACIÓN      73993832.0   

                                             TIPOALTA DIAGNOSTICO1  \
3946                                        DOMICILIO        I25.1   
9385                                        DOMICILIO        Z51.1   
9442                                        DOMICILIO        Z51.1   
9443            DERIVACIÓN OTRO HOSPITAL DEL SERVICIO        I21.3   
9539                                        DOMICILIO        I35.0   
...                                               ...          ...   
82470182                                    DOMICILIO        U07.1   
82470668                                    DOMICILIO        I05.9   
82471078  DERIVACIÓN OTRO HOSPITAL DE LA RED NACIONAL        I21.9   
82471166                                    DOMICILIO        I21.4   
82471581                                    DOMICILIO        I25.1   

          IR_29301_PESO  IR_29301_SEVERIDAD         variable procedimiento  
3946             0.8838                 1.0   PROCEDIMIENTO1         88.55  
9385             1.9970                 3.0   PROCEDIMIENTO1         99.25  
9442             0.9168                 2.0   PROCEDIMIENTO1         99.25  
9443             1.7114                 0.0   PROCEDIMIENTO1         00.66  
9539             3.2340                 1.0   PROCEDIMIENTO1         35.21  
...                 ...                 ...              ...           ...  
82470182         1.5219                 3.0  PROCEDIMIENTO30           NaN  
82470668         7.7014                 3.0  PROCEDIMIENTO30           NaN  
82471078         2.4840                 3.0  PROCEDIMIENTO30           NaN  
82471166         1.7184                 1.0  PROCEDIMIENTO30           NaN  
82471581         3.9336                 3.0  PROCEDIMIENTO30           NaN  

[403980 rows x 10 columns]

Se realizó el filtro de los procedimientos, solamente dejando los del hospital del Tórax.

Ahora, para obtener una cartera de procedimientos por diagnósticos se realizarán los siguientes
pasos (similar al realizado en el ámbito ambulatorio):

1. Se obtendrá el conteo de procedimientos por diagnóstico
2. Se obtendrá la cantidad de pacientes únicos por diagnóstico
3. Finalmente, se obtendrá la proporción de procedimientos por paciente para cada uno de los
diagnósticos.

In [15]:
canasta_procedimientos_hospitalizados = (
    (build_features.obtener_cartera_de_procedimientos_por_diagnostico(procedimientos_torax_long))
    .merge(cie_9, how="left", left_on="procedimiento", right_on="Código")
    .merge(cie_10, how="left", left_on="DIAGNOSTICO1", right_on="Código")
)

In [18]:
display(
    canasta_procedimientos_hospitalizados.query(
        "DIAGNOSTICO1 == 'C34.1' and procedimiento == '99.29'"
    )
)

ANIO_EGRESO DIAGNOSTICO1 procedimiento  cantidad_procedimientos  \
977         2019.0        C34.1         99.29                      175   
14337       2020.0        C34.1         99.29                       41   
23654       2021.0        C34.1         99.29                       52   

       cantidad_pacientes_distintos  cantidad_proced_por_pacientes Código_x  \
977                             176                       0.994318    99.29   
14337                            70                       0.585714    99.29   
23654                            60                       0.866667    99.29   

                                           Descripción_x  \
977    INYECCION SUSTANCIA TERAPEUTICA O PROFILACTICA...   
14337  INYECCION SUSTANCIA TERAPEUTICA O PROFILACTICA...   
23654  INYECCION SUSTANCIA TERAPEUTICA O PROFILACTICA...   

                                  Categoría_x  \
977    99 OTROS PROCEDIMIENTOS NO QUIRURGICOS   
14337  99 OTROS PROCEDIMIENTOS NO QUIRURGICOS   
23654  99 OTROS PROCEDIMIENTOS NO QUIRURGICOS   

                                               Sección_x  \
977    16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
14337  16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   
23654  16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...   

                                              Capítulo_x    Versión Código_y  \
977    16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...  CIE-v2013    C34.1   
14337  16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...  CIE-v2013    C34.1   
23654  16. (87 99) PROCEDIMIENTOS DIAGNOSTICOS Y TERA...  CIE-v2013    C34.1   

                                           Descripción_y  \
977    Tumor maligno del lóbulo superior, bronquio o ...   
14337  Tumor maligno del lóbulo superior, bronquio o ...   
23654  Tumor maligno del lóbulo superior, bronquio o ...   

                                           Categoría_y  \
977    C34 TUMOR MALIGNO DE LOS BRONQUIOS Y DEL PULMÓN   
14337  C34 TUMOR MALIGNO DE LOS BRONQUIOS Y DEL PULMÓN   
23654  C34 TUMOR MALIGNO DE LOS BRONQUIOS Y DEL PULMÓN   

                                               Sección_y  \
977    C30-C39  NEOPLASIAS MALIGNAS DE ÓRGANOS RESPIR...   
14337  C30-C39  NEOPLASIAS MALIGNAS DE ÓRGANOS RESPIR...   
23654  C30-C39  NEOPLASIAS MALIGNAS DE ÓRGANOS RESPIR...   

                         Capítulo_y  
977    Cap.02  NEOPLASIAS (C00-D49)  
14337  Cap.02  NEOPLASIAS (C00-D49)  
23654  Cap.02  NEOPLASIAS (C00-D49)

In [19]:
canasta_proceds_diags_mas_relevantes = canasta_procedimientos_hospitalizados[
    canasta_procedimientos_hospitalizados["DIAGNOSTICO1"].isin(DIAGNOSTICOS_MAS_RELEVANTES)
]

In [20]:
with pd.ExcelWriter("../data/interim/canasta_procedimientos_hospitalizados.xlsx") as file:
    canasta_procedimientos_hospitalizados.to_excel(
        file, sheet_name="canastas_globales", index=False
    )
    canasta_proceds_diags_mas_relevantes.to_excel(
        file, sheet_name="canasta_diags_mas_relevantes", index=False
    )

Los resultados muestran los procedimientos realizados para el diagnóstico C34.1.

- En primera lugar, se observa que el procedimiento más frecuente es el 90.59, realizado 283 veces. 
- Para el diagnóstico C34.1, existen 302 pacientes distintos entre 2019 y 2021.
- Finalmente, un paciente con un diagnóstico C34.1 se realiza en promedio 0.94 procedimientos 90.59.

Algo importante a tener en cuenta es que un mismo paciente se pudo haber realizado 10 veces
un procedimiento 90.59. Sin embargo, este análisis solamente da el promedio de procedimientos por
pacientes.

Algo interesante a responder sería saber cuál es la distribución en la cantidad de procedimientos
por diagnóstico. En otras palabras, responder las siguientes preguntas:

- ¿Cuántos procedimientos X se realiza un paciente que tiene el diagnóstico X? (Ya está respondido 
en el último apartado)
- **¿Cuántos es el máximo/mínimo de procedimientos X que se ha realizado un paciente que tiene el
diagnóstico X?**

A modo de control, se observará si el diagnóstico C34.1 efectivamente tiene 
283 procedimientos 90.59 y 302 pacientes distintos.

In [35]:
control_procedimientos = df_procesada.query(
    "COD_HOSPITAL == @HOSPITAL_DEL_TORAX and DIAGNOSTICO1 == 'C34.1'"
)

display(control_procedimientos)

COD_HOSPITAL  CIP_ENCRIPTADO    SEXO FECHA_NACIMIENTO    ETNIA  \
16208          112103       1129042.0  HOMBRE       1943-10-27  NINGUNO   
16246          112103       1101478.0  HOMBRE       1961-04-12  NINGUNO   
16257          112103        281602.0   MUJER       1936-02-06  NINGUNO   
16262          112103        641101.0   MUJER       1945-03-05  NINGUNO   
18845          112103       1166456.0  HOMBRE       1950-05-02  NINGUNO   
...               ...             ...     ...              ...      ...   
2678748        112103      68618209.0   MUJER       1949-12-08  NINGUNO   
2680142        112103      84741484.0  HOMBRE       1950-12-18  NINGUNO   
2690273        112103      69853991.0   MUJER       1944-10-26  NINGUNO   
2694816        112103      67743046.0   MUJER       1939-09-03  NINGUNO   
2700876        112103      67739218.0   MUJER       1947-10-25  NINGUNO   

         PROVINCIA      COMUNA NACIONALIDAD PREVISION  \
16208     SANTIAGO       ÑUÑOA        CHILE  FONASA B   
16246     SANTIAGO       MAIPU        CHILE  FONASA D   
16257     SANTIAGO       ÑUÑOA        CHILE  FONASA B   
16262      LINARES     LINARES        CHILE  FONASA B   
18845    COIHAIQUE   COIHAIQUE        CHILE  FONASA B   
...            ...         ...          ...       ...   
2678748  CACHAPOAL    RANCAGUA        CHILE  FONASA B   
2680142   SANTIAGO  SAN MIGUEL        CHILE  FONASA D   
2690273   SANTIAGO       MACUL        CHILE  FONASA B   
2694816   SANTIAGO  LAS CONDES        CHILE  FONASA B   
2700876   SANTIAGO   PEÑALOLÉN        CHILE  FONASA B   

                  SERVICIO_SALUD  ... IR_29301_MORTALIDAD  \
16208      METROPOLITANO ORIENTE  ...                 1.0   
16246      METROPOLITANO CENTRAL  ...                 3.0   
16257      METROPOLITANO ORIENTE  ...                 3.0   
16262                  DEL MAULE  ...                 3.0   
18845                      AYSEN  ...                 1.0   
...                          ...  ...                 ...   
2678748  LIBERTADOR B. O HIGGINS  ...                 1.0   
2680142        METROPOLITANO SUR  ...                 1.0   
2690273    METROPOLITANO ORIENTE  ...                 2.0   
2694816    METROPOLITANO ORIENTE  ...                 1.0   
2700876    METROPOLITANO ORIENTE  ...                 1.0   

                                           HOSPPROCEDENCIA ESTANCIA  \
16208                                                  NaN      0.0   
16246    HOSPITAL CLÍNICO METROPOLITANO EL CARMEN DOCTO...     16.0   
16257        HOSPITAL DEL SALVADOR (SANTIAGO, PROVIDENCIA)      1.0   
16262                                                  NaN     11.0   
18845                        HOSPITAL REGIONAL (COIHAIQUE)     11.0   
...                                                    ...      ...   
2678748                                                NaN      7.0   
2680142                                                NaN      5.0   
2690273                                                NaN      4.0   
2694816                                                NaN      6.0   
2700876                                                NaN      3.0   

        ANIO_EGRESO MES_EGRESO    FECHA RANGO_ETARIO cod_comuna codregion  \
16208        2019.0        2.0   2019-2     (70, 80]      13120        13   
16246        2019.0        3.0   2019-3     (50, 60]      13119        13   
16257        2019.0        3.0   2019-3     (80, 90]      13120        13   
16262        2019.0        1.0   2019-1     (70, 80]       7401         7   
18845        2019.0        6.0   2019-6     (60, 70]      11101        11   
...             ...        ...      ...          ...        ...       ...   
2678748      2021.0        5.0   2021-5     (70, 80]       6101         6   
2680142      2021.0        6.0   2021-6     (70, 80]      13130        13   
2690273      2021.0       12.0  2021-12     (70, 80]      13118        13   
2694816      2021.0        3.0   2021-3     (80, 90]      13114    

En primer lugar, se observa que el diagnóstico C34.1 ha tenido 367 egresos entre 2019 y 2021.

In [36]:
control_pacientes_distintos = (control_procedimientos.CIP_ENCRIPTADO.unique())
print(f"Para el diagnostico C34.1 hay {len(control_pacientes_distintos)} pacientes distintos")

Para el diagnostico C34.1 hay 302 pacientes distintos


Por lo tanto, efectivamente existen 302 pacientes distintos para el diagnóstico C34.1.

Solamente queda saber si se realizaron 283 procedimientos 90.59.

In [37]:
control_suma_procedimientos = (control_procedimientos[vars_procedimientos] == "90.59").sum()
display(control_suma_procedimientos)
print(f"Para el diagnostico C34.1 hay {control_suma_procedimientos.sum()} procedimientos 90.59")

PROCEDIMIENTO1      2
PROCEDIMIENTO2      4
PROCEDIMIENTO3      8
PROCEDIMIENTO4      8
PROCEDIMIENTO5     15
PROCEDIMIENTO6     14
PROCEDIMIENTO7     14
PROCEDIMIENTO8     18
PROCEDIMIENTO9     16
PROCEDIMIENTO10    10
PROCEDIMIENTO11    17
PROCEDIMIENTO12    15
PROCEDIMIENTO13    14
PROCEDIMIENTO14    17
PROCEDIMIENTO15    15
PROCEDIMIENTO16    13
PROCEDIMIENTO17    16
PROCEDIMIENTO18     7
PROCEDIMIENTO19    10
PROCEDIMIENTO20     9
PROCEDIMIENTO21     6
PROCEDIMIENTO22    11
PROCEDIMIENTO23     5
PROCEDIMIENTO24     5
PROCEDIMIENTO25     2
PROCEDIMIENTO26     4
PROCEDIMIENTO27     6
PROCEDIMIENTO28     1
PROCEDIMIENTO29     0
PROCEDIMIENTO30     1
dtype: int64

Para el diagnostico C34.1 hay 283 procedimientos 90.59


Por lo tanto, efectivamente se realizaron 283 procedimiento 90.59.

**Por ende, el análisis de canasta de procedimientos se ha realizado correctamente.**